theforeman.org/scripts/debian/stage_packages.sh

```diff
@@ -7,19 +7,44 @@ set -xe
 # * the web node must have the freight class applied
 
 # Find the deps in the build-dir from the previous step
-DEB_PATH=./debian/${os}/build-${project}
+DEB_PATH=$(find ./ -name build-${project})
 
 # Upload all builds to stagingdeb for testing
-echo "scratch build: uploading to stagingdeb/${os}/${repoowner}-${version}"
-export RSYNC_RSH="ssh -i /home/jenkins/workspace/staging_key/rsync_freightstage_key"
-USER=freightstage
 HOSTS="web01.osuosl"
-COMPONENT=${repoowner}-${version}
+
+if [[ "$DEB_PATH" == *"plugins"* ]];then
+  # Don't upload scratch builds to deb.tf.o
+  if [ x$repoowner = xtheforeman ] && [ x$pr_number = x ]; then
+    echo "Built from main repo, uploading to deb/plugins/main"
+    export RSYNC_RSH="ssh -i /home/jenkins/workspace/deb_key/rsync_freight_key"
+    USER=freight
+    if [ x$repo = xdevelop ]; then
+      COMPONENT=nightly
+    else
+      COMPONENT=$repo
+    fi
+  else
+    echo "scratch build: uploading to stagingdeb/plugins/${repoowner}"
+    export RSYNC_RSH="ssh -i /home/jenkins/workspace/staging_key/rsync_freightstage_key"
+    USER=freightstage
+    COMPONENT=${repoowner}
+  fi
+else
+  echo "scratch build: uploading to stagingdeb/${os}/${repoowner}-${version}"
+  export RSYNC_RSH="ssh -i /home/jenkins/workspace/staging_key/rsync_freightstage_key"
+  USER=freightstage
+  HOSTS="web01.osuosl"
+  COMPONENT=${repoowner}-${version}
+fi
 
 for HOST in $HOSTS; do
   # The path is important, as freight_rsync (which is run on the web node for incoming
   # transfers) will parse the path to figure out the repo to send debs to.
-  TARGET_PATH="${USER}@${HOST}.theforeman.org:rsync_cache/${os}/${COMPONENT}/"
+  if [[ "$DEB_PATH" == *"plugins"* ]];then
+    TARGET_PATH="${USER}@${HOST}.theforeman.org:rsync_cache/plugins/${COMPONENT}/"
```

Comment:
While we're at it, can we make `$HOST` the FQDN and simply use `$HOST` here? I feel like that's clearer for everyone and doesn't limit us to `theforeman.org`.